xlml/utils/tpu.py

```diff
@@ -372,3 +386,38 @@ def clean_up_idle_queued_resources(project_name: str, zones: Iterable[str]) -> N
       ):
         logging.info(f'Deleting {qr.name} in {state.name} status.')
         client.delete_queued_resource(name=qr.name)
+
+
+@task
+def clean_up_idle_nodes(project_name: str, zones: Iterable[str]) -> None:
+  """Clean up TPU nodes that are expired.
+
+  Args:
+   project_name: The project of resources.
+   zones: Available zones to clean up for the project.
+  """
+  creds, _ = google.auth.default()
+  client = tpu_api.TpuClient(credentials=creds)
+
+  logging.info(f'Cleaning up nodes in project {project_name}.')
+  for zone in zones:
+    logging.info(f'Checking in zone {zone.value}.')
+    parent = f'projects/{project_name}/locations/{zone.value}'
+    request = tpu_api.types.ListNodesRequest(parent=parent)
+    responses = client.list_nodes(request)
+
+    for node in responses:
+      create_time = node.create_time
+      current_time = datetime.datetime.now(datetime.timezone.utc)
+      ttl = int(node.labels[TTL]) if node.labels[TTL] else 0
+      if ttl != 0:
+        create_time = node.create_time
+        current_time = datetime.datetime.now(datetime.timezone.utc)
+        logging.info(
+            f'create_time is {create_time}, and current_time is {current_time}'
+        )
+        if (current_time - create_time).seconds >= ttl:
+          logging.info(
+              f'Deleting node {node.name} since ttl ({ttl} in secondes) is expired.'
```

Comment:
This is almost always going to be a large number of seconds. What do you think of using the `str` representation of `timedelta` to make this log more human-readable? https://stackoverflow.com/a/538721

also nit: `seconds`

Reply:
Yeah, I agree with readability. Let me have a try today. Thanks!

Reply:
Updated to make it clear. Let me know HDYT.